### ISSUE TITLE
Change file save system to hash based directory structure

### ISSUE THREAD

**Issue by Segelzwerg** [Segelzwerg/FamilyFoto#83]
```
images
 - ab
   - ab67df90ca173720982
     -  file (<- file with hash ab67df90ca173720982)
     -  thumb (<- thumbnail of file)
 - f9
   - f967df90ca173720982
     - file (<- another file with hash f967df90ca173720982)
     - thumb (<- thumbnail of above file)
```
proposed by @iTitus

**Comment by iTitus** [comment]
Or save every file under its hash. 

Git uses a slightly modified SHA1.

**Comment by Segelzwerg** [comment]
closed by #84
